### PR TITLE
Test setup

### PR DESCRIPTION
**Pull request by jgraff2**
organized by commit.

### CODE REVIEW

File: prototype/connection/testConnection.js [hunk 1]
```diff
@@ -43,13 +43,13 @@ describe('connection tests >', () => {
...
-   *   });
-   * });
-   */
+  describe('prepareUrl >', () => {
+    it('prepareUrl', () => {
+      const url = tu.prepareUrl(ctx);
+      expect(url).to.have.protocol('https');
+    });
+  });
+  */
```
Comment (shriramshankar) [inline prototype/connection/testConnection.js]: Why did you remove the asterisk in the multiline comment?

File: src/compressUtils.js [hunk 2]
```diff
@@ -53,8 +52,24 @@ const uglifyOpts = {
 };
 
 function buildConnection(dir = cwd) {
+  const name = path.basename(dir);
   const connectionPath = path.resolve(dir, 'connection', 'connection.js');
   const connectionExports = require(connectionPath);
+  const ctxDef = connectionExports.contextDefinition;
+
+  const f = path.resolve(dir, `${name}.json`);
+  return fs.readJson(f)
```
Comment (shriramshankar) [inline src/compressUtils.js]: Any reason behind using the async version of (read/write)Json, unlike the sync ones used in the tests? I think, if we use the sync version, we can avoid promise chaining in ` bin/build*` files. We also will not have to wrap `checkConflictingCtxDefs` with a promise.

Comment (jgraff2) [reply src/compressUtils.js]: Good point. I changed it to use sync methods for fs access. It's definitely cleaner this way.

@iamigo

File: prototype/connection/testConnection.js (hunk 1 above)
Comment (jgraff2) [reply prototype/connection/testConnection.js]: I changed it for consistency with the other files, which I changed to multiline comments before I realized this file already had it. I think either way is fine, but I prefer it without asterisks because it keeps the indentation the same whether it's commented or not.

File: src/compressUtils.js (hunk 2 above)
Comment (shriramshankar) [reply src/compressUtils.js]: @jgraff2  Sorry. I did not realize that my comment would cause so many other changes. We could have done it as a part of another story too.
